src/Build/Logging/TerminalLogger/TerminalProjectInfo.cs

```diff
@@ -7,6 +7,18 @@
 
 namespace Microsoft.Build.Logging;
 
+/// <summary>
+/// A struct containing relevant evaluation-time data that may not be knowable just from ProjectStart events.
+/// </summary>
+/// <param name="context"></param>
+/// <param name="ProjectFile"></param>
+/// <param name="TargetFramework"></param>
+/// <param name="RuntimeIdentifier"></param>
+internal record struct EvalProjectInfo(TerminalLogger.EvalContext context, string? ProjectFile, string? TargetFramework, string? RuntimeIdentifier)
+{
+    public readonly int Id => context.Id;
+}
```

Comment:
Since we're getting data from evals now, and evals are unique from projectcontexts/projectinstances, we need a way to capture what we're grabbing from the eval. This captures the state that used to belong to the TerminalProjectInfo (which is really mapped to a ProjectContextId) into a more shareable structure.